pkg/config/config.go

```diff
@@ -213,15 +216,23 @@ type Blueprint struct {
 	TerraformBackendDefaults TerraformBackend  `yaml:"terraform_backend_defaults"`
 }
 
+// ModConnection defines details about connections between modules. Currently,
+// only modules connected with "use" are tracked.
+type ModConnection struct {
+	toID   string
+	unused bool
+}
+
 // DeploymentConfig is a container for the imported YAML data and supporting data for
 // creating the blueprint from it
 type DeploymentConfig struct {
 	Config Blueprint
 	// Indexed by Resource Group name and Module Source
 	ModulesInfo map[string]map[string]modulereader.ModuleInfo
 	// Maps module ID to group index
-	ModuleToGroup map[string]int
-	expanded      bool
+	ModuleToGroup     map[string]int
+	expanded          bool
+	moduleConnections map[string][]ModConnection
```

Comment:
Normally storing graph like data is either done as:
1. nodes as objects, edges as pointers
2. list of edges
3. matrix (ignored in this case)

What it looks like we are doing here is kind of #1 but pointers being a data structure that contains the name of the next node. I am wondering if #2 might be a better representation here. In this case you could just have a list of connections which contain both the up and down stream id (as suggested elsewhere). 

I think just storing as a list would simplify the operations done in this CL. I am not sure if it would make other future operations harder.

Thoughts? This might just be personal preference bleeding through.

Reply:
Looking at it again with fresh eyes, you're absolutely right, I made the change and it simplified a lot of the code.

It might be helpful to access connections for all fromID/toID modules quickly at some point, but that can be added later.